pkg/identityserver/gateway_registry.go

```diff
@@ -254,6 +286,19 @@ func (is *IdentityServer) listGateways(ctx context.Context, req *ttnpb.ListGatew
 		if rights.RequireGateway(ctx, gtw.GatewayIdentifiers, ttnpb.RIGHT_GATEWAY_INFO) != nil {
 			gtws.Gateways[i] = gtw.PublicSafe()
 		}
+
+		if rights.RequireGateway(ctx, gtw.GatewayIdentifiers, ttnpb.RIGHT_GATEWAY_READ_SECRETS) != nil {
```

Comment:
Let's not validate the rights on every gateway unless the caller actually requests this field.